### PR TITLE
Buckify breeze

### PR DESCRIPTION
**Pull request by Yuhta**
Differential Revision: D65797319

### CODE REVIEW

**Comment by aditi-pandit** [comment]
@Yuhta : Just curious. What is breeze ?

**Comment by Yuhta** [comment]
@aditi-pandit Breeze is a new library developed by Rivos for cross platform SIMT acceleration. We keep it inside Velox so that it can be co-evolved with Velox.

**Comment by Yuhta** [comment]
CC: @dreveman

File: velox/experimental/wave/common/tests/CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +27,7 @@ add_executable(
 add_test(velox_wave_common_test velox_wave_common_test)
 set_tests_properties(velox_wave_common_test PROPERTIES LABELS cuda_driver)
 
+target_include_directories(velox_wave_common_test PRIVATE ../../../breeze)
```
Comment (majetideepak) [inline velox/experimental/wave/common/tests/CMakeLists.txt:30]: Better to use a header-only library for CMake.
https://discourse.cmake.org/t/header-only-library/5476

Comment (Yuhta) [reply velox/experimental/wave/common/tests/CMakeLists.txt:30]: I will leave this to @dreveman to encapsulate the library in a cmake project

File: velox/experimental/breeze/breeze/platforms/openmp.h [hunk 2]
```diff
@@ -235,7 +235,7 @@ struct OpenMPPlatform {
   inline unsigned lower_rank_lanemask() {
     static_assert(WARP_THREADS <= sizeof(unsigned) * 8,
                   "WARP_THREADS must be less or equal to unsigned bits");
-    return (1 << lane_idx()) - 1;
+    return (1u << lane_idx()) - 1;
```
Comment (dreveman) [inline velox/experimental/breeze/breeze/platforms/openmp.h:238]: can you move this change to a separate commit that has a short commit message explaining the need for it?

File: velox/experimental/breeze/breeze/utils/types.h [hunk 3]
```diff
@@ -166,7 +166,7 @@ class BadDeviceAlloc : public std::exception {
                  ",free=" + std::to_string(free) +
                  ",total=" + std::to_string(total) + ")") {}
 
-  virtual const char *what() const throw() { return message_.c_str(); }
+  virtual const char *what() const noexcept { return message_.c_str(); }
```
Comment (dreveman) [inline velox/experimental/breeze/breeze/utils/types.h:169]: same here. separate commit would be nice for this.

File: velox/experimental/breeze/test/generated/algorithms/algorithm_test-openmp.h [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Copyright (c) 2024 by Rivos Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
```
Comment (dreveman) [inline velox/experimental/breeze/test/generated/algorithms/algorithm_test-openmp.h:20]: Now that we're checking in this code. Let's make the generator code add a comment here that makes it clear that this file was auto-generated and should not be edited directly. Something like:

```
/*
 * This file is auto-generated from test/generate.sh
 * DO NOT EDIT!
 */
```

File: velox/experimental/wave/common/tests/CMakeLists.txt (hunk 1 above)
Comment (dreveman) [reply velox/experimental/wave/common/tests/CMakeLists.txt:30]: Yep, sounds like a good thing to do as a follow up.

**Review by dreveman — APPROVED** [review]
lgtm. just some suggestions but they can also be done as follow ups

File: velox/experimental/breeze/test/generated/algorithms/algorithm_test-openmp.h (hunk 4 above)
Comment (Yuhta) [reply velox/experimental/breeze/test/generated/algorithms/algorithm_test-openmp.h:20]: Sorry only saw this comment after merging, it's a good idea, can you add it to #11537?

**Pull request closed without merging**